juju-tutorials/how-to-fixing-attach-volume-to-a-k8s-instance.md

```diff
@@ -0,0 +1,336 @@
+3. Initialize the vault deployment.
+   ```bash
+   export VAULT_ADDR="http://$(juju status | grep vault/0 | tr -s " " | cut -d" " -f5):8200"
+   vault operator init -key-shares=5 -key-threshold=3 > vault-keys
+   ```
+4. Unseal all keys and generate root CA.
+   ```bash
+   for key in $(cat vault-keys | grep -E "Unseal Key .*: " | cut -c15-)
+   do 
+       vault operator unseal $key
+   done
+   ```
+5. Authorize the vault unit.
+   ```bash
+   export VAULT_TOKEN=$(cat vault-keys | grep -E "Initial Root Token: " | cut -c21-)
+   vault token create -ttl=10m
+   juju run-action --wait vault/leader authorize-charm token=$VAULT_TOKEN
+   juju run-action --wait vault/leader generate-root-ca
+   ```
+   
+##### add external L3 connectivity
+
+1. Get OpenStack admin net.
+   ```bash
+   export ADMIN_NET=$(openstack network list | grep admin_net | cut -d"|" -f3 | tr -d " ")
+   ```
+2. Create new port for each nova-compute unit. (by default there are 3 units)
+   ```bash
+   unset BRIDGE_INTERFACE_MAPPING  # clear value
+
+   for i in 0 1 2
+   do
+       # create new port
+       export NEW_PORT_$i_ID=$(openstack port create --network $ADMIN_NET data-port | grep ' id ' | cut -d"|" -f3 | tr -d " ")
+       echo "  create new port with id $NEW_PORT_$i_ID"
+    
+       # get ovn chassis ip
+       export OVN_CHASSIS_$i_IP=$(juju status | grep ovn-chassis/$i | tr -s " " | cut -d" " -f5)
+       echo "  found ovn-chassis/$i ip address $OVN_CHASSIS_$i_IP"
+    
+       # get ovn chassis id
+       export OVN_CHASSIS_$i_ID=$(openstack server list | grep -E $OVN_CHASSIS_$i_IP | cut -d"|" -f3 | tr -d " ")
+       echo "  found ovn-chassis/$i id $OVN_CHASSIS_$i_ID"
+    
+       # add port to ovn chassis
+       openstack server add port $OVN_CHASSIS_$i_ID $NEW_PORT_$i_ID
+       echo "  add port $NEW_PORT_$i_ID to server $OVN_CHASSIS_$i_ID"
+    
+       # get port mac address
+       export NEW_PORT_$i_MAC_ADDRESS=$(openstack port show $NEW_PORT_$i_ID | grep mac_address | cut -d"|" -f3 | tr -d " ")
+       echo "  port $NEW_PORT_$i_ID mac address $NEW_PORT_$i_MAC_ADDRESS"
+    
+       # set bridge interface mapping variable
+       export BRIDGE_INTERFACE_MAPPING="$BRIDGE_INTERFACE_MAPPING br-ex:$NEW_PORT_$i_MAC_ADDRESS"
+   done
+   ```
+3. Configure the bridge interface mapping in the `ovn-chassis` application.
+   ```bash
+   juju config ovn-chassis bridge-interface-mappings="$BRIDGE_INTERFACE_MAPPING"
+   ```
+   
+##### final configuration
+
+The final step is to configure juju to use deployed OpenStack as cloud and configure network on OpenStack.
+
+[note status="OpenStack credentials"]
+Don't forget to get credentials for the new OpenStack. (e.g. `source openrcv3_project`)
+[/note]
+
+1. To access the OpenStack cloud is necessary to set up flavors, images, networks and security. Detailed instructions can be found on [openstack-base][OpenStackBase] in the "Accessing the cloud" section.
```

Comment:
"In order to use the managed OpenStack service, several configurations are required: flavors, images, networks and security groups. Detailed ..."